Source/Core/VideoCommon/TextureCacheBase.h

```diff
@@ -61,23 +37,31 @@ class TextureCacheBase
     u32 memory_stride;
     bool is_efb_copy;
     bool is_custom_tex;
-    bool may_have_overlapping_textures;
+    bool may_have_overlapping_textures = true;
 
     unsigned int native_width,
         native_height;  // Texture dimensions from the GameCube's point of view
     unsigned int native_levels;
 
     // used to delete textures which haven't been used for TEXTURE_KILL_THRESHOLD frames
-    int frameCount;
+    int frameCount = FRAMECOUNT_INVALID;
 
     // Keep an iterator to the entry in textures_by_hash, so it does not need to be searched when
     // removing the cache entry
-    std::multimap<u64, TCacheEntryBase*>::iterator textures_by_hash_iter;
+    std::multimap<u64, TCacheEntry*>::iterator textures_by_hash_iter;
 
     // This is used to keep track of both:
     //   * efb copies used by this partially updated texture
     //   * partially updated textures which refer to this efb copy
-    std::unordered_set<TCacheEntryBase*> references;
+    std::unordered_set<TCacheEntry*> references;
+
+    TCacheEntry(std::unique_ptr<AbstractTexture> texture) : texture(std::move(texture)) {}
```

Comment:
The constructor should be `explicit`.

The definition for the constructor and destructor should be within the cpp file.